core/src/main/java/nu/yona/server/messaging/service/MessageDto.java

```diff
@@ -165,6 +166,7 @@ public boolean isRead()
 		private SenderInfo.Factory senderInfoFactory;
 
 		@Override
+		@Transactional
```

Comment:
Can't this be put in `MessageService.TheDtoManager.handleAction`? It seems strange to put it in a DTO.

Reply:
It's not the DTO but the manager of the DTO. That's a ``Component`` (and always has been). That component needs to be transactionally safe, which is why the annotation belongs here. The ``MessageService.TheDtoManager.handleAction`` is already annotated with ``@Transactional``, but Spring basically suspends that transaction when calling a nontransactional component.